src/compute-client/src/controller/instance.rs

```diff
@@ -262,6 +262,16 @@ impl<T> Instance<T> {
         };
         self.ready_responses.push_back(resp);
     }
+
+    pub fn reverse_compute_dependencies(&self, id: GlobalId) -> impl Iterator<Item = &GlobalId> {
+        self.collections_iter().filter_map(move |(id2, state)| {
```

Comment:
nit: I'm not sure why the `move` is still needed here, you can skip it and do `contains(id)` where `id` will be borrowed as `&GlobalId`.

Reply:
The problem is that the closure is actually part of the return value, because of returning an Iterator, so if I remove `move`, then I get
```
error[E0373]: closure may outlive the current function, but it borrows `id`, which is owned by the current function
   --> src/compute-client/src/controller/instance.rs:268:44
    |
268 |         self.collections_iter().filter_map(|(dependant_id, state)| {
    |                                            ^^^^^^^^^^^^^^^^^^^^^^^ may outlive borrowed value `id`
269 |             if state.compute_dependencies.contains(&id) {
    |                                                     -- `id` is borrowed here
    |
note: closure is returned here
   --> src/compute-client/src/controller/instance.rs:268:9
    |
268 | /         self.collections_iter().filter_map(|(dependant_id, state)| {
269 | |             if state.compute_dependencies.contains(&id) {
270 | |                 Some(dependant_id)
271 | |             } else {
272 | |                 None
273 | |             }
274 | |         })
    | |__________^
help: to force the closure to take ownership of `id` (and any other referenced variables), use the `move` keyword
    |
268 |         self.collections_iter().filter_map(move |(dependant_id, state)| {
    |                                            ++++

```

Reply:
Urgh...